### DISCUSSION TITLE
Can't handle checked exceptions using an interceptor/annotation

### DISCUSSION THREAD

**Discussion by hydrowoxy** [quarkusio/quarkus#44517] [Q&A] [answered]
Hello,

I am trying to use AOP to simplify exception handling for file storage operations and eliminate repetitive try-catch logic. The goal is to use an interceptor to catch a custom `StorageException` (which extends `RuntimeException`) and `IOException`. The idea is conceptually similar to this: https://medium.com/@ashitcmd/simplified-exception-handling-aop-in-spring-applications-63aa7976139d

I have defined an interceptor which catches `IOException` and `StorageException` using `@AroundInvoke`. I have also defined an annotation `OnStorageException` which binds the interceptor. 

When I inject `@OnStorageException` over a method that could throw a `StorageException` and `IOException`, it handles the `StorageException` just fine, but tells me that I'm leaving `IOException` unhandled. It does catch the `IOException` in the interceptor but the compiler still thinks it is unhandled. I assume this is because `IOException` is a checked exception while `StorageException` extends `RuntimeException` and so is unchecked.

One fix is to declare that the method `throws IOException` but this is messy as I have to change the signatures in the interface where it is defined as well. Is there any way to handle checked errors within the interceptor, without needing to explicitly declare them in the method itself?

Thanks in advance.

**Comment by HerrDerb** [comment, marked as answer]
I don't think this is possible by the way how java works. With AOP or generally interceptors you cannot modify the signature of a method. The given Spring example targets a RestController. You never call those methods directly in the code (the framework does over reflection) that is why the compiler does not complain.

You might want to checkout lomboks `@SneakyThrows`:https://www.baeldung.com/java-sneaky-throws
But this pretty much rewrites your code with a preprocessor -> https://github.com/projectlombok/lombok/blob/master/src/core/lombok/SneakyThrows.java#L39

**Reply by hydrowoxy** [reply]
Thank you! using `@SneakyThrows` + the interceptor seems to solve the problem. Appreciate the help.
